mp4parse/src/lib.rs

```diff
@@ -1714,45 +1714,38 @@ fn read_stsd<T: Read>(src: &mut BMFFBox<T>, track: &mut Track) -> Result<SampleD
 fn read_sinf<T: Read>(src: &mut BMFFBox<T>) -> Result<ProtectionSchemeInfoBox> {
     let mut sinf = ProtectionSchemeInfoBox::default();
 
-    let mut has_frma_schi_boxes = (false, false);
     let mut iter = src.box_iter();
     while let Some(mut b) = iter.next_box()? {
         match b.head.name {
             BoxType::OriginalFormatBox => {
                 let frma = read_frma(&mut b)?;
                 sinf.code_name = frma;
-                has_frma_schi_boxes.0 = true;
             },
             BoxType::SchemeInformationBox => {
                 // We only need tenc box in schi box so far.
-                let schi_tenc = read_schi(&mut b)?;
-                sinf.tenc = schi_tenc;
-                has_frma_schi_boxes.1 = true
+                sinf.tenc = read_schi(&mut b)?;
             },
             _ => skip_box_content(&mut b)?,
         }
         check_parser_state!(b.content);
     }
 
-    if has_frma_schi_boxes != (true, true) {
-        return Err(Error::InvalidData("malformat sinf box"));
-    }
-
     Ok(sinf)
 }
 
-fn read_schi<T: Read>(src: &mut BMFFBox<T>) -> Result<TrackEncryptionBox> {
+fn read_schi<T: Read>(src: &mut BMFFBox<T>) -> Result<Option<TrackEncryptionBox>> {
+    let mut tenc = None;
     let mut iter = src.box_iter();
     while let Some(mut b) = iter.next_box()? {
         match b.head.name {
             BoxType::TrackEncryptionBox => {
-                return read_tenc(&mut b);
+                tenc = Some(read_tenc(&mut b)?);
```

Comment:
Should we do something (log a message and/or error) if there are multiple tenc boxes?